inference-explainers/inference_onboarding_guide.md

```diff
@@ -0,0 +1,228 @@
+
+## Cloud Deployment
+
+The Bidding and Auction services can be deployed on Google Cloud Platform (GCP) or Amazon Web Services (AWS). We provide GCP and AWS support for using Bidding and Auction services with inference capabilities. Documentation for packaging and deploying Bidding and Auction services is available in the [Bidding and Auction services GCP cloud support and deployment guide][3] (GCP) and the [Bidding and Auction services AWS cloud support and deployment guide][4] (AWS).
+
+The inference capabilities require additional steps to configure cloud storage and Terraform-based services, and are described in this document. While inference for Bidding and Auction services is designed to be generalized to other services, inference support is currently limited to the Bidding service.
+
+To deploy and use the inference capabilities, you will:
+
+- Build and package the Bidding and Auction servers images.
+- Upload ML models to a cloud bucket and generate a metadata file summarizing all models you are using.
+- Update UDF code modules to invoke the inference JavaScript functions.
+- Supply additional inference-specific Terraform configurations to deploy the Bidding and Auction servers with inference enabled.
+
+The following sections provide a step-by-step breakdown of the cloud deployment process for each supported cloud.
+
+### GCP Guide: Package, deploy, and run a service
+
+This section describes how to package and deploy B&A services on GCP.
+
+The process for creating a functioning service with inference on GCP has two major steps:
+
+1. **Packaging:** Create a Docker image containing the services’ code.
+2. **Deployment:** Run Terraform to bring up the Bidding and Auction services with inference.
+
+#### Step 0: Prerequisites
+
+1. Refer to the [Bidding and Auction services GCP cloud support and deployment guide][3] to familiarize yourself with the packaging and deployment process for GCP. This document assumes you are able to create a functioning service stack (following Step 0 to Step 2.5 in the Package, Deploy, and Run a Service Guide section of the above guide). It highlights the additional steps required to enable inference capabilities.
+2. Running Bidding and Auction services with inference requires functioning machine learning models. Choose Tensorflow or PyTorch as your ML platform, and then familiarize yourself with the [Tensorflow][5] or [PyTorch][6] ML framework. This document assumes you are able to create and save models in your chosen framework.
+
+#### Step 1: Packaging
+
+1. **Build the GCP Confidential Space Docker image:** Follow the instructions in the aforementioned [GCP Guide][3] to build a docker image for all B&A services. The inference artifacts, including both the “tensorflow_v2_14_0” and “pytorch_v2_1_1” (ML runtimes and versions that we currently support) sidecar binaries, will be packaged within the Bidding service image. <br/>
```

Comment:
Can we move sidecar to earlier in the sentence, like this:

 **Build the GCP Confidential Space Docker image:** Follow the instructions in the aforementioned [GCP Guide][3] to build a docker image for all B&A services. The inference artifacts, including both the “tensorflow_v2_14_0” and “pytorch_v2_1_1”  *sidecar binaries* (ML runtimes and versions that we currently support), will be packaged within the Bidding service image. <br/>

Reply:
Done